bathbot/src/commands/osu/rank/pp.rs

```diff
@@ -803,12 +803,25 @@ impl RankData {
                     pp_missing(user_pp, rank_holder_pp, scores)
                 };
 
+                let idx = idx + 1;
+
+                let idx_suffix = match idx % 100 {
+                    11 | 12 | 13 => "th",
+                    _ => match idx % 10 {
+                        1 => "st",
+                        2 => "nd",
+                        3 => "rd",
+                        _ => "th",
+                    },
+                };
```

Comment:
This can be refactored into its own function so it can be re-used. It's header could look something like
```rs
fn idx_suffix(idx: usize) -> &'static str { ... }
```